ignite/contrib/handlers/time_profilers.py

```diff
@@ -437,3 +438,395 @@ def odict_to_str(d: Mapping):
+        t = self._event_handlers_timer.value()
+        hname = hname = self._get_callable_name(func)
+        self.event_handlers_times[event][hname][e] = t
+
+    def _timeit_iteration_events(self, handler: Callable, engine: Engine, event: Events, *args, **kwargs):
+        # wrapper for profiling handlers attached to iteration events
+        self._event_handlers_timer.reset()
+        func, args, kwargs = handler
+        func(*args, **kwargs)
+        i = engine.state.iteration - 1
+        t = self._event_handlers_timer.value()
+        hname = self._get_callable_name(func)
+        self.event_handlers_times[event][hname][i] = t
+
+    def _timeit_processing(self, engine: Engine):
+        # handler used for profiling processing times
+        t = self._processing_timer.value()
+        i = engine.state.iteration - 1
+        self.processing_times[i] = t
+
+    def _timeit_dataflow(self, engine: Engine):
+        # handler used for profiling dataflow times
+        t = self._dataflow_timer.value()
+        i = engine.state.iteration - 1
+        self.dataflow_times[i] = t
+
+    def _reset(self, num_epochs: int, total_num_iters: int):
+        # reset the variable used for profiling
+        self.dataflow_times = torch.zeros(total_num_iters)
+        self.processing_times = torch.zeros(total_num_iters)
+
+        self.event_handlers_times = {
+            e: {h: torch.zeros(1) for h in self.event_handlers_names[e]} for e in self._global_events
+        }
+
+        self.event_handlers_times.update(
+            {e: {h: torch.zeros(num_epochs) for h in self.event_handlers_names[e]} for e in self._epoch_events}
+        )
+
+        self.event_handlers_times.update(
+            {e: {h: torch.zeros(total_num_iters) for h in self.event_handlers_names[e]} for e in self._iteration_events}
+        )
+
+    @staticmethod
+    def _is_internal_handler(handler: Callable):
+        # checks whether the handler is internal
+        return "HandlersTimeProfiler." in repr(handler)
+
+    def _detach_profiler_handlers(self, engine):
+        # reverts handlers to original handlers
+        engine._event_handlers = self._orig_event_handlers
+
+    def _as_first_started(self, engine):
+        if hasattr(engine.state.dataloader, "__len__"):
+            num_iters_per_epoch = len(engine.state.dataloader)
+        else:
+            num_iters_per_epoch = engine.state.epoch_length
+
+        self.max_epochs = engine.state.max_epochs
+        self.total_num_iters = self.max_epochs * num_iters_per_epoch
+
+        self.event_handlers_names = {
+            e: [
+                h.__qualname__ if hasattr(h, "__qualname__") else h.__class__.__name__
+                for (h, _, _) in engine._event_handlers[e]
+                if not self._is_internal_handler(h)
+            ]
+            for e in self._events
+        }
+
+        self._reset(self.max_epochs, self.total_num_iters)
+
+        self._orig_event_handlers = engine._event_handlers.copy()
```

Comment:
I'm afraid of copying any type objects like that and not sure if it would work as expected. As event handlers can contain function and its args/kwargs where user can pass everything he/she wants.